`std.mem.zeroes` does not work with `std.PackedIntArray` error: cannot assign to constant

### Zig Version

0.9.0-dev.1277+c4df9bf56

### Steps to Reproduce

```zig
const std = @import("std");
test "foo" {
    _ = std.mem.zeroes(std.PackedIntArray(u2, 4));
}
```

Note: `git bisect` indicates that this was introduced with commit 526191bfafe722700323df30647eed0c03fc2403

### Expected Behavior

Compile without error.

### Actual Behavior

```
➜  bugs zig test zeroes.zig
/Users/ehaas/source/zig/build/lib/zig/std/mem.zig:258:59: error: cannot assign to constant
                    @field(structure, field.name) = zeroes(@TypeOf(@field(structure, field.name)));
                                                          ^
./zeroes.zig:3:23: note: called from here
    _ = std.mem.zeroes(std.PackedIntArray(u2, 4));
                      ^
./zeroes.zig:2:15: note: called from here
test "zeroes" {
              ^
```

--- a/lib/std/mem.zig
+++ b/lib/std/mem.zig
@@ -344,6 +344,8 @@ test "mem.zeroes" {
     try testing.expect(a.y == 10);
 
     const ZigStruct = struct {
+        comptime comptime_field: u8 = 5,
+        
         integral_types: struct {
             integer_0: i0,
             integer_8: i8,
@@ -378,6 +380,7 @@ test "mem.zeroes" {
     };
 
     const b = zeroes(ZigStruct);
+    try testing.expectEqual(@as(u8, 5), b.comptime_field);
     try testing.expectEqual(@as(i8, 0), b.integral_types.integer_0);
     try testing.expectEqual(@as(i8, 0), b.integral_types.integer_8);
     try testing.expectEqual(@as(i16, 0), b.integral_types.integer_16);

--- a/lib/std/mem.zig
+++ b/lib/std/mem.zig
@@ -275,7 +275,9 @@ pub fn zeroes(comptime T: type) T {
             } else {
                 var structure: T = undefined;
                 inline for (struct_info.fields) |field| {
-                    @field(structure, field.name) = zeroes(@TypeOf(@field(structure, field.name)));
+                    if (!field.is_comptime) {
+                        @field(structure, field.name) = zeroes(@TypeOf(@field(structure, field.name)));
+                    }
                 }
                 return structure;
             }

--- a/lib/std/mem.zig
+++ b/lib/std/mem.zig
@@ -345,7 +345,7 @@ test "mem.zeroes" {
 
     const ZigStruct = struct {
         comptime comptime_field: u8 = 5,
-        
+
         integral_types: struct {
             integer_0: i0,
             integer_8: i8,

